Page object commands not working in 3.0.1 typescript projects

### Description of the bug/issue

When I use a typescript page object command in a Nightwatch test project I expect to be able to explicitly specific the page object's interface as the type and use commands defined in the page object in command chaining, but instead I receive error 

```console
Type 'EnhancedPageObject<any, any, any>' is not assignable to type 'EcosiaPage'.
  Type 'EnhancedPageObject<any, any, any>' is not assignable to type 'EnhancedPageObjectSharedFields<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error | null>; }>, Required<{ [name: string]: string; } | { ...; } | { ...; } | undefined>, null, {}, string>'.
    Types of property 'elements' are incompatible.
      Type '{ [x: string]: EnhancedElementInstance<EnhancedPageObject<Required<any>, Required<any>, any, {}, string>>; }' is not assignable to type '{ [x: string]: EnhancedElementInstance<EnhancedPageObject<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error | null>; }>, Required<{ [name: string]: string; }>, null, {}, string>>; } | { ...; } | { ...; } | undefined'.
        Type '{ [x: string]: EnhancedElementInstance<EnhancedPageObject<Required<any>, Required<any>, any, {}, string>>; }' is not assignable to type '{ [x: 
```
and

```console
Property 'clickSearch' does not exist on type 'Awaitable<NightwatchAPI, null>'.ts(2339)
```
![image](https://github.com/nightwatchjs/nightwatch/assets/18431256/32617b97-82df-47b3-b7bb-ef04c8559c25)

![image](https://github.com/nightwatchjs/nightwatch/assets/18431256/b610c7fc-ffeb-4a38-a1de-1ad86737cef7)



### Steps to reproduce

1. npm init nightwatch@latest

(as of writing that should setup Nightwatch 3.0.1)

2. Create a folder under /nightwatch called /page-objects
3. Create ecosiaPage.ts
```ts
// nightwatch/page-objects/ecosiaPage.ts

import { EnhancedPageObject, PageObjectModel } from 'nightwatch';

const ecosiaCommands = {
  clickSearch(this: EcosiaPage) {
    return this.waitForElementVisible('@submit', 10000)
      .click('@submit')
      .waitForElementNotPresent('@submit');
  },
};

const ecosiaPage: PageObjectModel = {
  url: 'http://www.ecosia.com',
  commands: [ecosiaCommands],
  elements: {
    searchInput: {
      selector: 'input[type=search]',
    },
    submit: {
      selector: 'button[type=submit]',
    },
  },
};

export default ecosiaPage;

export interface EcosiaPage
  extends EnhancedPageObject<
    typeof ecosiaCommands,
    typeof ecosiaPage.elements,
    null
  > {}
```

4. In the tests folder create ecosia.ts

```ts
import { EcosiaPage } from '../page-objects/ecosiaPage';

describe('Ecosia.org Demo', function () {
  this.tags = ['demo'];

  before((browser) => browser.navigateTo('https://www.ecosia.org/'));

  it('Demo test ecosia.org', function (browser) {
    const ecosia: EcosiaPage = browser.page.ecosiaPage();

    // Can call clickSearch here without IDE errors
    ecosia.clickSearch();

    ecosia
      .waitForElementVisible('body')
      .assert.titleContains('Ecosia')
      .assert.visible('@searchInput')
      .setValue('@searchInput', 'nightwatch')
      // Can't use it chained
      .clickSearch()
      .assert.not.textContains('.layout__content', 'Nightwatch.js');
  });

  after((browser) => browser.end());
});
```

5. Update nightwatch.conf.js settings

```js
  src_folders: ['nightwatch/tests'],
  page_objects_path: ['nightwatch/page-objects'],
```

6. Create ./nightwatch/types/page-objects.d.ts

```ts
import { NightwatchCustomPageObjects } from 'nightwatch';
import { EcosiaPage } from '../page-objects/ecosiaPage';

declare module 'nightwatch' {
  interface NightwatchCustomPageObjects {
    ecosiaPage(): EcosiaPage;
  }
}
```

7. In the inner nightwatch/tsconfig.json add `"include": ["types"]` below compilerOptions
8. In the other ./tsconfig.json add  `"include": ["./nightwatch/types"]`

(I'm not sure if 7 and 8 are necessary, but was following steps from a prior bug ticket)

9. Run `npx nightwatch -t nightwatch\tests\ecosia.ts`

Receive error:

```console
  TSError
   ⨯ Unable to compile TypeScript:
nightwatch/tests/ecosia.ts:9:11 - error TS2322: Type 'EnhancedPageObject<any, any, any>' is not assignable to type 'EcosiaPage'.
  Type 'EnhancedPageObject<any, any, any>' is not assignable to type 'EnhancedPageObjectSharedFields<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error>; }>, Required<{ [name: string]: string; } | { [name: string]: ElementProperties; } | { ...; }>, null, {}, string>'.
    Types of property 'elements' are incompatible.
      Type '{ [x: string]: EnhancedElementInstance<EnhancedPageObject<Required<any>, Required<any>, any, {}, string>>; }' is not assignable to type '{ [x: string]: EnhancedElementInstance<EnhancedPageObject<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error>; }>, Required<{ [name: string]: string; }>, null, {}, string>>; } | { ...; } | { ...; }'.
        Type '{ [x: string]: EnhancedElementInstance<EnhancedPageObject<Required<any>, Required<any>, any, {}, string>>; }' is not assignable to type '{ [x: string]: EnhancedElementInstance<EnhancedPageObject<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error>; }>, Required<{ [x: string]: ElementProperties; }>, null, {}, string>>; }'.
          'string' index signatures are incompatible.
            Type 'EnhancedElementInstance<EnhancedPageObject<Required<any>, Required<any>, any, {}, string>>' is not assignable to type 'EnhancedElementInstance<EnhancedPageObject<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error>; }>, Required<{ [x: string]: ElementProperties; }>, null, {}, string>>'.
              Type 'EnhancedPageObject<Required<any>, Required<any>, any, {}, string>' is not assignable to type 'EnhancedPageObject<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error>; }>, Required<{ [x: string]: ElementProperties; }>, null, {}, 
string>'.
                Type 'EnhancedPageObject<Required<any>, Required<any>, any, {}, string>' is not assignable to type 'EnhancedPageObjectSharedFields<Required<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error>; }>>, Required<Required<{ [x: string]: 
ElementProperties; }>>, null, {}, string>'.
                  Types of property 'section' are incompatible.
                    Type '{ [x: string]: EnhancedSectionInstance<Required<any>, Required<any>, Required<any>, Required<any>>; }' is not assignable to type 'null'.

9     const ecosia: EcosiaPage = browser.page.ecosiaPage();
            ~~~~~~
nightwatch/tests/ecosia.ts:16:8 - error TS2339: Property 'clickSearch' does not exist on type 'Awaitable<NightwatchAPI, null>'.

16       .clickSearch()
          ~~~~~~~~~~~
```
Might also be worth noting if you don't specific the test and choose to run all you get no output at all (which seems like another issue)
```console
npx nightwatch
Running   default: ecosia.ts  
 Wrote HTML report file to: C:\Users\Mr\Desktop\New folder\tests_output\nightwatch-html-report\index.html
```

### Sample test

```node
import { EcosiaPage } from '../page-objects/ecosiaPage';

describe('Ecosia.org Demo', function () {
  this.tags = ['demo'];

  before((browser) => browser.navigateTo('https://www.ecosia.org/'));

  it('Demo test ecosia.org', function (browser) {
    const ecosia: EcosiaPage = browser.page.ecosiaPage();

    // Can call clickSearch here without IDE errors
    ecosia.clickSearch();

    ecosia
      .waitForElementVisible('body')
      .assert.titleContains('Ecosia')
      .assert.visible('@searchInput')
      .setValue('@searchInput', 'nightwatch')
      // Can't use it chained
      .clickSearch()
      .assert.not.textContains('.layout__content', 'Nightwatch.js');
  });

  after((browser) => browser.end());
});
```


### Command to run

```bash
npx nightwatch -t nightwatch\tests\ecosia.ts
```


### Verbose Output

```fundamental
npx nightwatch -t nightwatch\tests\ecosia.ts --verbose
 Wrote HTML report file to: C:\Users\Mr\Desktop\New folder\tests_output\nightwatch-html-report\index.html

 Wrote Rerun Json report file to: C:\Users\Mr\Desktop\New folder\tests_output\minimal_report.json
  TSError
   ⨯ Unable to compile TypeScript:
nightwatch/tests/ecosia.ts:9:11 - error TS2322: Type 'EnhancedPageObject<any, any, any>' is not assignable to type 'EcosiaPage'.
  Type 'EnhancedPageObject<any, any, any>' is not assignable to type 'EnhancedPageObjectSharedFields<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error>; }>, Required<{ [name: string]: string; } | { [name: string]: ElementProperties; } | { ...; }>, null, {}, string>'.
    Types of property 'elements' are incompatible.
      Type '{ [x: string]: EnhancedElementInstance<EnhancedPageObject<Required<any>, Required<any>, any, {}, string>>; }' is not assignable to type '{ [x: string]: EnhancedElementInstance<EnhancedPageObject<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error>; }>, Required<{ [name: string]: string; }>, null, {}, string>>; } | { ...; } | { ...; }'.
        Type '{ [x: string]: EnhancedElementInstance<EnhancedPageObject<Required<any>, Required<any>, any, {}, string>>; }' is not assignable to type '{ [x: string]: EnhancedElementInstance<EnhancedPageObject<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error>; }>, Required<{ [x: string]: ElementProperties; }>, null, {}, string>>; }'.
          'string' index signatures are incompatible.
            Type 'EnhancedElementInstance<EnhancedPageObject<Required<any>, Required<any>, any, {}, string>>' is not assignable to type 'EnhancedElementInstance<EnhancedPageObject<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error>; }>, Required<{ [x: string]: ElementProperties; }>, null, {}, string>>'.
              Type 'EnhancedPageObject<Required<any>, Required<any>, any, {}, string>' is not assignable to type 'EnhancedPageObject<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error>; }>, Required<{ [x: string]: ElementProperties; }>, null, {}, 
string>'.
                Type 'EnhancedPageObject<Required<any>, Required<any>, any, {}, string>' is not assignable to type 'EnhancedPageObjectSharedFields<Required<Required<{ clickSearch(this: EcosiaPage): Awaitable<EcosiaPage, Error>; }>>, Required<Required<{ [x: string]: 
ElementProperties; }>>, null, {}, string>'.
                  Types of property 'section' are incompatible.
                    Type '{ [x: string]: EnhancedSectionInstance<Required<any>, Required<any>, Required<any>, Required<any>>; }' is not assignable to type 'null'.

9     const ecosia: EcosiaPage = browser.page.ecosiaPage();
            ~~~~~~
nightwatch/tests/ecosia.ts:16:8 - error TS2339: Property 'clickSearch' does not exist on type 'Awaitable<NightwatchAPI, null>'.

16       .clickSearch()
          ~~~~~~~~~~~
```


### Nightwatch Configuration

```node
// Refer to the online docs for more details:
// https://nightwatchjs.org/gettingstarted/configuration/
//

//  _   _  _         _      _                     _          _
// | \ | |(_)       | |    | |                   | |        | |
// |  \| | _   __ _ | |__  | |_ __      __  __ _ | |_   ___ | |__
// | . ` || | / _` || '_ \ | __|\ \ /\ / / / _` || __| / __|| '_ \
// | |\  || || (_| || | | || |_  \ V  V / | (_| || |_ | (__ | | | |
// \_| \_/|_| \__, ||_| |_| \__|  \_/\_/   \__,_| \__| \___||_| |_|
//             __/ |
//            |___/

module.exports = {
  // An array of folders (excluding subfolders) where your tests are located;
  // if this is not specified, the test source must be passed as the second argument to the test runner.
  src_folders: ['nightwatch/tests'],
  page_objects_path: ['nightwatch/page-objects'],

  // See https://nightwatchjs.org/guide/extending-nightwatch/adding-custom-commands.html
  custom_commands_path: [],

  // See https://nightwatchjs.org/guide/extending-nightwatch/adding-custom-assertions.html
  custom_assertions_path: [],

  // See https://nightwatchjs.org/guide/extending-nightwatch/adding-plugins.html
  plugins: [],

  // See https://nightwatchjs.org/guide/concepts/test-globals.html
  globals_path: '',

  webdriver: {},

  test_workers: {
    enabled: true,
  },

  test_settings: {
    default: {
      disable_error_log: false,
      launch_url: 'http://localhost',

      screenshots: {
        enabled: false,
        path: 'screens',
        on_failure: true,
      },

      desiredCapabilities: {
        browserName: 'chrome',
      },

      webdriver: {
        start_process: true,
        server_path: '',
      },
    },

    chrome: {
      desiredCapabilities: {
        browserName: 'chrome',
        'goog:chromeOptions': {
          // More info on Chromedriver: https://sites.google.com/a/chromium.org/chromedriver/
          //
          // w3c:false tells Chromedriver to run using the legacy JSONWire protocol (not required in Chrome 78)
          w3c: true,
          args: [
            //'--no-sandbox',
            //'--ignore-certificate-errors',
            //'--allow-insecure-localhost',
            //'--headless'
          ],
        },
      },

      webdriver: {
        start_process: true,
        server_path: '',
        cli_args: [
          // --verbose
        ],
      },
    },
  },
};
```


### Nightwatch.js Version

3.0.1

### Node Version

18.16.0

### Browser

Chrome 114.0.5735.134

### Operating System

Windows 10

### Additional Information

[nightwatch.zip](https://github.com/nightwatchjs/nightwatch/files/11839308/nightwatch.zip)

--- a/types/page-object.d.ts
+++ b/types/page-object.d.ts
@@ -1,9 +1,14 @@
 import {MergeObjectsArray} from './utils';
 import {NightwatchCustomCommands} from './custom-command';
 import {
+  AlertsNsCommands,
+  Assert,
   ChromiumClientCommands,
+  CookiesNsCommands,
+  DocumentNsCommands,
   ElementCommands,
   ElementFunction,
+  Expect,
   LocateStrategy,
   NamespacedApi,
   NightwatchAPI,
@@ -167,8 +172,7 @@ export type EnhancedSectionInstance<
     | 'clickAndHold'
     | 'doubleClick'
     | 'rightClick'
-  > &
-  Pick<NamespacedApi<NightwatchAPI>, 'alerts' | 'cookies' | 'document' | 'assert' | 'verify' | 'expect'>;
+  >;
 
 interface PageObjectSection {
   commands?: Record<string, unknown>[];
@@ -254,6 +258,14 @@ interface EnhancedPageObjectSharedFields<
    * Nightwatch new element API.
    */
   element: ElementFunction;
+
+  // Namespaces directly available on page-objects and sections.
+  alerts: AlertsNsCommands<this>;
+  cookies: CookiesNsCommands<this>;
+  document: DocumentNsCommands<this>;
+  assert: Assert<this>;
+  verify: Assert<this>;
+  expect: Expect;
 }
 
 export interface ElementProperties {
@@ -457,7 +469,6 @@ export type EnhancedPageObject<
   URL = string
 > = SharedCommands &
   NightwatchCustomCommands &
-  Pick<NamespacedApi<NightwatchAPI>, 'alerts' | 'cookies' | 'document' | 'assert' | 'verify' | 'expect'> &
   EnhancedPageObjectSharedFields<
     Required<MergeObjectsArray<Commands>>,
     Required<MergeObjectsArray<Elements>>,

--- a/types/tests/index.test-d.ts
+++ b/types/tests/index.test-d.ts
@@ -1,4 +1,4 @@
-import { expectError, expectType } from 'tsd';
+import { expectAssignable, expectError, expectType } from 'tsd';
 import { EventEmitter } from 'events';
 import {
   EnhancedPageObject,
@@ -414,7 +414,7 @@ const menuSection = {
 } satisfies SectionProperties;
 
 const googleCommands = {
-  submit(this: EnhancedPageObject) {
+  submit(this: GooglePage) {
     this.api.pause(1000);
     return this.waitForElementVisible('@submitButton', 1000)
       .click('@submitButton')
@@ -481,10 +481,18 @@ declare module '..' {
 const testPage = {
   'Test commands': () => {
     const google = browser.page.google();
-    google.setValue('@searchBar', 'nightwatch').submit();
+    google.setValue('@searchBar', 'nightwatch').submit().assert.titleContains('nightwatch');
 
     expectType<NightwatchAPI>(google.api);
     expectType<NightwatchClient>(google.client);
+
+    const result = google
+      .setValue('@searchBar', 'nightwatch')
+      .assert.titleContains('Google');
+    
+    expectAssignable<GooglePage>(result);
+    expectAssignable<GooglePage>(result.submit());
+    expectAssignable<GooglePage>(result.cookies.getAll());
 
     // test new element api
     google.element('@searchBar');
@@ -496,19 +504,21 @@ const testPage = {
   'Test sections': () => {
     const google = browser.page.google();
 
-    expectType<Awaitable<NightwatchAPI, null>>(google.cookies.deleteAll());
+    expectAssignable<GooglePage>(google.cookies.deleteAll());
     expectError(googlePage.window.maximize());
 
     const menuSection = google.section.menu;
 
-    menuSection
+    const result = menuSection
       .assert.visible('@mail')
       .assert.visible('@images');
+
+    expectAssignable<typeof menuSection>(result);
 
     menuSection.expect.element('@mail').to.be.visible;
     menuSection.expect.element('@images').to.be.visible;
 
-    expectType<Awaitable<NightwatchAPI, null>>(menuSection.alerts.accept());
+    expectAssignable<typeof menuSection>(menuSection.alerts.accept());
     expectType<NightwatchAPI>(menuSection.api);
     expectType<NightwatchClient>(menuSection.client);
 

